daemon/api_model.go

```diff
@@ -21,25 +21,45 @@ package daemon
 
 import (
 	"encoding/json"
+	"errors"
 	"net/http"
+	"net/url"
 
 	"github.com/snapcore/snapd/asserts"
 	"github.com/snapcore/snapd/overlord/auth"
 	"github.com/snapcore/snapd/overlord/devicestate"
 )
 
-var modelCmd = &Command{
-	Path: "/v2/model",
-	POST: postModel,
-	// TODO: provide GET here too once we decided on the details of the API
-}
+var (
+	serialModelCmd = &Command{
+		Path: "/v2/model/serial",
+		GET:  getSerial,
+	}
+	modelCmd = &Command{
+		Path: "/v2/model",
+		POST: postModel,
+		GET:  getModel,
```

Comment:
and this should have a `UserOK: true` as well

Reply:
I thought that it wasn't okay for a user to do a POST on the /v2/model endpoint, as that would allow any user to remodel the device which AFAICT is undesirable.

Reply:
UserOK etc only apply to GET; non-GET is always auth'ed

Reply:
Okay, thanks

Reply:
Added for both